FFmpeg 9 removed AVVulkanDeviceContext queue_family_* / nb_*_queues
fields in favor of qf/nb_qf.

--- a/gui/src/qmlmainwindow.cpp
+++ b/gui/src/qmlmainwindow.cpp
@@ -1375,14 +1375,28 @@
     vkctx->nb_enabled_inst_extensions = placebo_vk_inst->num_extensions;
     vkctx->enabled_dev_extensions = placebo_vulkan->extensions;
     vkctx->nb_enabled_dev_extensions = placebo_vulkan->num_extensions;
-    vkctx->queue_family_index = placebo_vulkan->queue_graphics.index;
-    vkctx->nb_graphics_queues = placebo_vulkan->queue_graphics.count;
-    vkctx->queue_family_tx_index = placebo_vulkan->queue_transfer.index;
-    vkctx->nb_tx_queues = placebo_vulkan->queue_transfer.count;
-    vkctx->queue_family_comp_index = placebo_vulkan->queue_compute.index;
-    vkctx->nb_comp_queues = placebo_vulkan->queue_compute.count;
-    vkctx->queue_family_decode_index = vk_decode_queue_index;
-    vkctx->nb_decode_queues = 1;
+    vkctx->nb_qf = 0;
+    vkctx->qf[vkctx->nb_qf++] = {
+        .idx = static_cast<int>(placebo_vulkan->queue_graphics.index),
+        .num = static_cast<int>(placebo_vulkan->queue_graphics.count),
+        .flags = VK_QUEUE_GRAPHICS_BIT,
+    };
+    vkctx->qf[vkctx->nb_qf++] = {
+        .idx = static_cast<int>(placebo_vulkan->queue_transfer.index),
+        .num = static_cast<int>(placebo_vulkan->queue_transfer.count),
+        .flags = VK_QUEUE_TRANSFER_BIT,
+    };
+    vkctx->qf[vkctx->nb_qf++] = {
+        .idx = static_cast<int>(placebo_vulkan->queue_compute.index),
+        .num = static_cast<int>(placebo_vulkan->queue_compute.count),
+        .flags = VK_QUEUE_COMPUTE_BIT,
+    };
+    vkctx->qf[vkctx->nb_qf++] = {
+        .idx = vk_decode_queue_index,
+        .num = 1,
+        .flags = VK_QUEUE_VIDEO_DECODE_BIT_KHR,
+    };
+#if FF_API_VULKAN_SYNC_QUEUES
     vkctx->lock_queue = [](struct AVHWDeviceContext *dev_ctx, uint32_t queue_family, uint32_t index) {
         auto vk = reinterpret_cast<pl_vulkan>(dev_ctx->user_opaque);
         vk->lock_queue(vk, queue_family, index);
@@ -1391,6 +1405,7 @@
         auto vk = reinterpret_cast<pl_vulkan>(dev_ctx->user_opaque);
         vk->unlock_queue(vk, queue_family, index);
     };
+#endif
     if (av_hwdevice_ctx_init(vulkan_hw_dev_ctx) < 0) {
         qCWarning(chiakiGui) << "Failed to create Vulkan decode context";
         av_buffer_unref(&vulkan_hw_dev_ctx);
